sdk/python/tests/test_sanity_delete.py

```diff
@@ -0,0 +1,144 @@
+#  ----------------------------------------------------------------
+# Copyright 2016 Cisco Systems
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ------------------------------------------------------------------
+"""
+    test_sanity_delete.py
+        Unittest for DELETE object.
+"""
+import ydk.types as ytypes
+import unittest
+
+from ydk.services import CRUDService
+from ydk.providers import NetconfServiceProvider
+from ydk.types import Empty, DELETE, Decimal64
+from tests.compare import is_equal
+from ydk.errors import YPYError, YPYDataValidationError
+from ydk.models.ydktest import ydktest_sanity as ysanity
+
+class SanityYang(unittest.TestCase):
+
+    @classmethod
+    def setUpClass(self):
+        self.ncc = NetconfServiceProvider(
+            address='127.0.0.1', username='admin', 
+            password='admin', protocol='ssh', port=12022)
+        self.crud = CRUDService()
+
+    @classmethod
+    def tearDownClass(self):
+        self.ncc.close()
+
+    def setUp(self):
+        runner = ysanity.Runner()
+        self.crud.delete(self.ncc, runner)
+        print '\nIn method', self._testMethodName + ':'
+
+    def tearDown(self):
+        runner = ysanity.Runner()
+        self.crud.delete(self.ncc, runner)
+
+    def test_delete_object_on_leaf(self):
+        # create runner with two leaves, one and two
+        runner_create = ysanity.Runner()
+        runner_create.one.name = 'one'
+        runner_create.two.name = 'two'
+        self.crud.create(self.ncc, runner_create)
+
+        runner_read_filter = ysanity.Runner()
+        runner_read = self.crud.read(self.ncc, runner_read_filter)
+
+        # use DELETE object to remove leaf one
+        runner_delete = runner_read
+        runner_delete.one.name = DELETE()
+        self.crud.update(self.ncc, runner_delete)
+
+        # manually create remaining runner with leaf two
+        runner_read = self.crud.read(self.ncc, runner_read_filter)
+        runner_left = runner_create
+        runner_left.one.name = None
+
+        self.assertEqual(is_equal(runner_read, runner_left), True)
+
+    @unittest.skip('DELETE object for leaf-list')
```

Comment:
Yes, I think so. Tried locally and the validator complains about this DELETE object. The `_has_data` function used by this leaf-list's parent(and parent's parent, etc) tries to iterate through elements of this leaf-list, which is assigned to a DELETE object(TypeError). And I think, we might need to get value of elements in leaf-list(which is shadowed by this DELETE object in this use case), since payload accepted by confd is:
```
  <edit-config>
    <target>
      <candidate/>
    </target>
    <config xmlns:xc="urn:ietf:params:xml:ns:netconf:base:1.0">
      <runner xmlns="http://cisco.com/ns/yang/ydktest-sanity">
        <one>
          <name>one</name>
        </one>
        <ytypes>
          <built-in-t>
            <llstring xc:operation="delete">0</llstring>
            <llstring xc:operation="delete">1</llstring>
          </built-in-t>
        </ytypes>
      </runner>
    </config>
  </edit-config>
```
but not
```
  <edit-config>
    <target>
      <candidate/>
    </target>
    <config xmlns:xc="urn:ietf:params:xml:ns:netconf:base:1.0">
      <runner xmlns="http://cisco.com/ns/yang/ydktest-sanity">
        <one>
          <name>one</name>
        </one>
        <ytypes>
          <built-in-t>
            <llstring xc:operation="delete"/>
          </built-in-t>
        </ytypes>
      </runner>
    </config>
  </edit-config>
```